
Make mmake honnor the autoconf man, info and doc dir variables.

Index: mercury-14.01.1/scripts/Mmake.vars.in
===================================================================
--- mercury-14.01.1.orig/scripts/Mmake.vars.in
+++ mercury-14.01.1/scripts/Mmake.vars.in
@@ -37,7 +37,7 @@ EXTRA_INIT_DIRS =
 MERCURY_EXTRA_INIT_DIRS	=
 endif
 
-# Set the directory search path. 
+# Set the directory search path.
 # (See the GNU Make manual for documentation about VPATH and GPATH.)
 # We need to substitute all spaces with colons for the VPATH to work.
 # Getting a space to be recognised as the first argument of the subst
@@ -193,7 +193,7 @@ EXTRA_MCSFLAGS	= $(EXTRA_MCFLAGS)
 #
 
 # FIX_PATH_FOR_<foo> should be set to a script which converts path names
-# from Unix/Cygwin format into a format suitable for the <foo> command.  
+# from Unix/Cygwin format into a format suitable for the <foo> command.
 # If <foo> is a Unix/Cygwin command that expects Unix-style path names,
 # FIX_PATH_FOR_<foo> should be set to "echo".
 # If <foo> is a Windows command that expects Windows-style path names,
@@ -225,7 +225,7 @@ MERC_DLL_DIR    = $(MERCURY_STDLIB_DIR)/
 # ILASM is the Microsoft IL assembler, which turns IL assembly code
 # into bytecode.
 # XXX we should also support the Portable.NET version of ILASM,
-# which has a different command-line interface. 
+# which has a different command-line interface.
 # (Is there a Mono ILASM too?  What about Rotor?)
 ILASM		= @ILASM@
 ALL_ILASMFLAGS= $(ILASMFLAGS) $(EXTRA_ILASMFLAGS) \
@@ -235,11 +235,11 @@ EXTRA_ILASMFLAGS =
 LIB_ILASMFLAGS= $(patsubst %,-I %,$(EXTRA_C_INCL_DIRS))
 
 # MS_AL is the Microsoft assembly linker, which creates assemblies.
-MS_AL		= @MS_AL@ 
+MS_AL		= @MS_AL@
 
 # CSC is the command line version of C# compiler
 # XXX we should also support the Portable.NET and Mono C# compilers,
-# which have a different command-line interface. 
+# which have a different command-line interface.
 # (What about Rotor?)
 
 FIX_PATH_FOR_CSC = @CYGPATH@
@@ -263,7 +263,7 @@ ALL_JAVACFLAGS	= $(JAVACFLAGS) $(EXTRA_J
 JAVACFLAGS	= @JAVAC_FLAGS_FOR_HEAP_SIZE@
 EXTRA_JAVACFLAGS =
 # XXX Should we set LIB_JAVACFLAGS?
-LIB_JAVACFLAGS	= 
+LIB_JAVACFLAGS	=
 
 # JAVA_INTERPRETER is the command line Java interpreter;
 # it should be able to run Java bytecode files.
@@ -381,8 +381,8 @@ EXT_FOR_PIC_OBJECTS	= @EXT_FOR_PIC_OBJEC
 EXT_FOR_SHARED_LIB	= @EXT_FOR_SHARED_LIB@
 
 # lists of targets that depend on $(ALL_MLOBJS) and $(ALL_MLPICOBJS)
-MLOBJS_DEPS = 
-MLPICOBJS_DEPS = 
+MLOBJS_DEPS =
+MLPICOBJS_DEPS =
 
 #-----------------------------------------------------------------------------#
 
@@ -682,25 +682,29 @@ else
     DESTDIR_AND_SLASH=$(DESTDIR)/
 endif
 
+# Weirdness between Mmake and autoconf
+prefix=@prefix@
+PACKAGE=mercury-$(VERSION)
+
 INSTALL_PREFIX		= $(DESTDIR_AND_SLASH)@prefix@
 INSTALL_BINDIR		= $(INSTALL_PREFIX)/bin
 INSTALL_LIBDIR		= $(INSTALL_PREFIX)/lib/mercury
-INSTALL_INFO_DIR	= $(INSTALL_PREFIX)/info
-INSTALL_DVI_DIR		= $(INSTALL_PREFIX)/lib/mercury/doc
-INSTALL_TEXT_DIR	= $(INSTALL_PREFIX)/lib/mercury/doc
-INSTALL_PS_DIR		= $(INSTALL_PREFIX)/lib/mercury/doc
-INSTALL_PDF_DIR		= $(INSTALL_PREFIX)/lib/mercury/doc
-INSTALL_MAN_DIR		= $(INSTALL_PREFIX)/man
-INSTALL_HTML_DIR	= $(INSTALL_PREFIX)/lib/mercury/html
-INSTALL_MDB_DOC_DIR	= $(INSTALL_PREFIX)/lib/mercury/mdb
+INSTALL_INFO_DIR	= $(DESTDIR_AND_SLASH)@infodir@
+INSTALL_DVI_DIR		= $(DESTDIR_AND_SLASH)@docdir@
+INSTALL_TEXT_DIR	= $(DESTDIR_AND_SLASH)@docdir@
+INSTALL_PS_DIR		= $(DESTDIR_AND_SLASH)@docdir@
+INSTALL_PDF_DIR		= $(DESTDIR_AND_SLASH)@docdir@
+INSTALL_MAN_DIR		= $(DESTDIR_AND_SLASH)@mandir@
+INSTALL_HTML_DIR	= $(DESTDIR_AND_SLASH)@docdir@
+INSTALL_MDB_DOC_DIR	= $(DESTDIR_AND_SLASH)@docdir@
 INSTALL_ELISP_DIR	= $(INSTALL_PREFIX)/lib/mercury/elisp
 INSTALL_CGI_DIR		= $(DESTDIR_AND_SLASH)@CGIDIR@
 
 FINAL_INSTALL_PREFIX	= @prefix@
 FINAL_INSTALL_BINDIR 	= $(FINAL_INSTALL_PREFIX)/bin
 FINAL_INSTALL_LIBDIR	= $(FINAL_INSTALL_PREFIX)/lib/mercury
-FINAL_INSTALL_INFO_DIR	= $(FINAL_INSTALL_PREFIX)/info
-FINAL_INSTALL_MAN_DIR	= $(FINAL_INSTALL_PREFIX)/man
+FINAL_INSTALL_INFO_DIR	= @infodir@
+FINAL_INSTALL_MAN_DIR	= @mandir@
 FINAL_INSTALL_ELISP_DIR	= $(FINAL_INSTALL_PREFIX)/lib/mercury/elisp
 FINAL_INSTALL_CGI_DIR	= @CGIDIR@
 
